doc/explanation/grouping_options.ipynb

```diff
@@ -0,0 +1,386 @@
+    "- **Vectorized coloring**: Each point colored by category\n",
+    "- **Cannot be indexed**: Single element, not separable by category\n",
+    "\n",
+    "### When to use:\n",
+    "- When you want the best performance for large datasets\n",
+    "- When you need smooth, continuous color mapping\n",
+    "- When you don't need to isolate specific categories"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f54ccbed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_color = penguins.hvplot.scatter(\n",
+    "    x='bill_length_mm',\n",
+    "    y='bill_depth_mm',\n",
+    "    color='species',\n",
+    "    title=\"Color: Vectorized Coloring\"\n",
+    ")\n",
+    "plot_color"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d4fb28a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Alternative syntax using 'c'\n",
+    "plot_c = penguins.hvplot.scatter(\n",
+    "    x='bill_length_mm',\n",
+    "    y='bill_depth_mm',\n",
+    "    c='species',\n",
+    "    title=\"Using 'c' parameter (same result)\"\n",
+    ")\n",
+    "plot_c"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "60b4c757",
+   "metadata": {},
+   "source": [
+    "### Indexing and access:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ab54c88e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Cannot index by category - this won't work:\n",
+    "print(f\"Type of 'color' plot: {type(plot_color)}\")\n",
+    "try:\n",
+    "    adelie_color = plot_color['Adelie']  # This will raise an error!\n",
+    "except Exception:\n",
+    "    print(\"Error: You cannot index a single element by category!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1626f2c0",
+   "metadata": {},
+   "source": [
+    "## Visual Output Comparison\n",
+    "\n",
+    "Let's compare all three approaches side by side to see the differences:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a1545149",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import panel as pn\n",
+    "\n",
+    "width = 300\n",
+    "\n",
+    "groupby_plot = penguins.hvplot.scatter(\n",
+    "    x='bill_length_mm', y='bill_depth_mm',\n",
+    "    groupby='species', title=\"groupby='species'\",\n",
+    "    frame_width=width, widget_location='bottom_right',\n",
+    ")\n",
+    "\n",
+    "by_plot = penguins.hvplot.scatter(\n",
+    "    x='bill_length_mm', y='bill_depth_mm',\n",
+    "    by='species', title=\"by='species'\",\n",
+    "    frame_width=width,\n",
+    ")\n",
+    "\n",
+    "color_plot = penguins.hvplot.scatter(\n",
+    "    x='bill_length_mm', y='bill_depth_mm',\n",
+    "    color='species', title=\"color='species'\",\n",
+    "    frame_width=width,\n",
+    ")\n",
+    "\n",
+    "pn.Column(pn.Row(by_plot, color_plot), groupby_plot)"
```

Comment:
Put groupby first? Like in the order of how they are documented. In fact all in a Column would work I think.